backend/services/theme_service.py

```
@@ -185,6 +185,69 @@ async def generate_with_face_reference(
+                if not parsed_response.get('data') or not parsed_response['data'][0].get('imageURL'):
+                    raise HTTPException(status_code=502, detail="RunWare API did not return an image URL for the variation.")
+
+                img_url = parsed_response['data'][0]['imageURL']
+                img_response = await client.get(img_url)
+                img_response.raise_for_status()
+                
+                logger.info("✅ Variation generated successfully.")
+                return img_response.content
+        except httpx.HTTPStatusError as e:
+            logger.error(f"❌ RunWare variation generation failed with status {e.response.status_code}: {e.response.text}")
+            raise HTTPException(status_code=e.response.status_code, detail=f"RunWare variation generation failed: {e.response.text}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)